data/types.go

```diff
@@ -208,11 +210,44 @@ func (f TargetFileMeta) HashAlgorithms() []string {
 }
 
 type Targets struct {
-	Type        string      `json:"_type"`
-	SpecVersion string      `json:"spec_version"`
-	Version     int         `json:"version"`
-	Expires     time.Time   `json:"expires"`
-	Targets     TargetFiles `json:"targets"`
+	Type        string       `json:"_type"`
+	SpecVersion string       `json:"spec_version"`
+	Version     int          `json:"version"`
+	Expires     time.Time    `json:"expires"`
+	Targets     TargetFiles  `json:"targets"`
+	Delegations *Delegations `json:"delegations,omitempty"`
+}
+
+type Delegations struct {
+	Keys  map[string]*Key `json:"keys"`
+	Roles []DelegatedRole `json:"roles"`
+}
+
+type DelegatedRole struct {
+	Name             string   `json:"name"`
+	KeyIDs           []string `json:"keyids"`
+	Threshold        int      `json:"threshold"`
+	PathHashPrefixes []string `json:"path_hash_prefixes,omitempty"`
+	Paths            []string `json:"paths"`
+	Terminating      bool     `json:"terminating"`
+}
+
+func (d DelegatedRole) MatchesPath(file string) bool {
```

Comment:
I updated to allow only one of `paths` or `path_hash_prefixes` at json decoding level https://github.com/theupdateframework/go-tuf/pull/137/commits/8eac49ac1dfc789722d85f5c53731a07a89e012d
This will err when decoding, and enforce in the DelegatedRole struct that both are never set together

Reply:
I opened a PR suggesting what I think is a cleaner way that doesn't require multiple structs, copying, or renaming: https://github.com/raphaelgavache/go-tuf/pull/3

Reply:
Thank you, I accepted your suggestions